### ISSUE TITLE
<edit-config> targeting candidate datastore doesn't postpone validation

### ISSUE THREAD

**Issue by syyyr** [CESNET/Netopeer2#431]
Hello,
I'm trying to send an `<edit-config>` operation to the Netopeer2-server targeting the candidate datastore through Netopeer2-cli like so:
```
> edit-config --target candidate --config=data.xml
```
data.xml:
```
<interfaces xmlns="urn:ietf:params:xml:ns:yang:ietf-interfaces" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0">
  <interface nc:operation="create">
    <name>eth2</name>
  </interface>
</interfaces>
```
When the operation is issued, this error is returned:
```
ERROR
	type:     application
	tag:      operation-failed
	severity: error
	path:     /ietf-interfaces:interfaces/interface[name='eth2']
	message:  Missing required element "type" in "interface".
```
I would like to be able to specify the `type` element in a different `<edit-config>` operation. From my understanding, validation of the candidate datastore should happen only after the `<commit>` operation is issued (as written here: https://tools.ietf.org/html/rfc7950#section-8.3.3).

(In my program, I'm actually using the libnetconf2 library to send the `<edit-config>` requests, but it appears the problem does not depend on the client.)

**Comment by michalvasko** [comment]
Hi,
you are right, thanks for referencing the section, we missed that piece of information.

Regards,
Michal
